internal/pkg/service/buffer/store/watcher/state.go

```diff
@@ -133,19 +134,21 @@ func watch[T fmt.Stringer](s *State, prefix etcdop.PrefixT[T]) *objectsState[T]
 
 	tree := prefixtree.NewWithLock[T](s.lock)
 	ch, initDone := watchFactory()
+
+	// Log only changes, not initial load
+	logsEnabled := atomic.NewBool(false)
+	go func() {
+		<-initDone
+		logsEnabled.Store(true)
+	}()
+
 	s.wg.Add(1)
 	go func() {
 		defer s.wg.Done()
-
-		// Log only changes, not initial load
-		logsEnabled := false
-
 		for {
 			select {
 			case <-s.ctx.Done():
 				return
-			case <-initDone:
-				logsEnabled = true
```

Comment:
Tu bol bug.

`initDone` signalizuje, ze sa nacitali uz vsetky hodnoty cez `getAll` a zacina sa `watch`.
Realizuje sa to pomocou zatvorenia `chan` `initDone`.

A teda, ked je `chan` zatvoreny, tak tento `case` vzdy prejde a sprava sa to ako `for {}` - vytazuje to CPU.